l10n_ch_invoice_reports/controllers/portal.py

```diff
@@ -0,0 +1,47 @@
+from odoo import http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+
+from odoo.addons.account.controllers.portal import PortalAccount
+
+
+class PortalAccount(PortalAccount):
+    @http.route(
+        ["/my/invoices/<int:invoice_id>"], type="http", auth="public", website=True
+    )
+    def portal_my_invoice_detail(
```

Comment:
Looks like you are redefining an existing route from the `account` module.
Hard to tell what changes but it seems to be the pdf download, only.
In that case it would be better/cleaner to not copy paste the existing implementation, but implement the download of the pdf here and use the parent implementation with `super` for the rest

Reply:
I agree w/ @TDu.
Also, when overriding a route method, don't re-define all params. Just use `@http.route()`

Reply:
here i am changing only parameter (report_ref) with which method is called depending on the type of document. It is not defined anywhere before. Not sure how super() can help. Already talked about that with @TDu on Friday.

http.route fixed.